torchao/float8/fsdp_utils.py

```diff
@@ -67,7 +67,7 @@ def precompute_float8_dynamic_scale_for_fsdp(module: nn.Module) -> None:
     scale_tensor = torch.finfo(torch.float8_e4m3fn).max / amax_tensor  # Replicate
     if amax_tensor.dtype is torch.float16:
         scale_tensor = torch.clamp(scale_tensor, max=torch.finfo(torch.float16).max)
-    local_scale_tensor = scale_tensor.to_local()
+    local_scale_tensor = scale_tensor.to_local().to(dtype=torch.float32)
```

Comment:
thanks for pinpoint to a specific line. I can also take a look at why `max_weights = torch._foreach_norm(weights, ord=math.inf)` does not ends up with `float32`. I was expecting local_scale_tensor to be the same dtype to `model.parameters()`

Reply:
just realized that we cast to float32 in https://github.com/pytorch/ao/blob/227d4bfece6bb00f0365642ad1d7afd509079ba4/torchao/float8/float8_utils.py#L55

it's good to keep precompute in sync with float8_utils.py. thanks for the fix